docs/04_resource_management/041_exception_handling.ipynb

```diff
@@ -0,0 +1,155 @@
+    "   \n",
+    "   Dies führt zu dieser Ausgabe:\n",
+    "   ```\n",
+    "   <class 'ZeroDivisionError'>\n",
+    "   division by zero\n",
+    "   ```\n",
+    "\n",
+    "Oft wirst du in die Situation kommen, dass du verschiedene Fehler verschieden oder nur bestimmte Fehler behandeln möchtest.\n",
+    "Hierfür bietet es sich an, verschiedene `except`-Blöcke zu definieren:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    number = int(input(\"Please give a number: \"))\n",
+    "    print(f\"100 / {number} = {100/number}\")\n",
+    "except ZeroDivisionError:\n",
+    "    print(\"Use another input than 0. Can't divide by 0.\")\n",
+    "except ValueError:\n",
+    "    print(\"Please enter a valid number!\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Nach dem `except`-Keyword haben wir den Typ der Exception (des Fehlers) angegeben. Der Inhalt des entsprechenden `except`-Blocks wird nur ausgeführt, wenn der Fehler vom spezifizierten Typ ist (bzw. davon erbt).\n",
+    "\n",
+    "Im allerersten `except` haben wir die Exception-Klasse `Exception` angegeben. Da die anderen Exception-Klassen wie `ZeroDivisionError` und `ValueError` eine Subklasse von `Exception` sind, werden diese Fehler auch von `except Exception` abgefangen.\n",
+    "\n",
+    "Die beiden genaueren Exceptions dienen für folgenden Zweck:\n",
+    "* `ZeroDivisionError`: Wird geworfen, wenn eine Zahl durch 0 geteilt wird.\n",
+    "* `ValueError`: Ein Parameter (hier derjenige für `input`) hat einen nicht unterstützen Wert erhalten."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Vollständiges Beispiel\n",
+    "Der Vollständigkeit halber präsentieren wir hier ein komplettes `try-except-else-finally`-Exception Handling:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "try:\n",
+    "    number = int(input(\"Please give a number: \"))\n",
+    "    result = 100/number\n",
+    "except ZeroDivisionError:\n",
+    "    print(\"Use another input than 0. Can't divide by 0.\")\n",
+    "except ValueError:\n",
+    "    print(\"Please enter a valid number!\")\n",
+    "else:\n",
+    "    print(\"The operation succeeded. The result:\", result)\n",
+    "finally:\n",
+    "    print(\"Now, let's go into vacations :)\")"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Der obige Code beinhaltet\n",
+    "* das ursprüngliche Error-Handling, das auf `ZeroDivisionError`- und `ValueError`-Fehler behandelt.\n",
+    "* einen `else`-Block, der ausgeführt wird, wenn es im `try` keinen Fehler gegeben hat.\n",
+    "* und einen `finally`-Block, der **immer** im Anschluss noch ausgeführt wird."
```

Comment:
Ich würde hier aufführen, dass "immer" in diesem Kontext bedeutet, dass dieser Block ausgeführt wird egal ob es einen Exception gegeben hat oder nicht. Damit ist es klarer.